### PR TITLE
PR: Add support for completion inside f-strings.

### LINKED ISSUE #7265: Option for detecting f strings.
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
### Enhancement proposal 

Hello,

It would be nice to have an option in Spyder which, when turned on, detect when '{' or '}' are written in string. It would then add a 'f' in front of the string in order to set up a formatted string. 

Hope to see this feature in Spyder one day. 

Thanks for your amazing work !

### PR DESCRIPTION
**Pull request by dalthviz**
### Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md)
* [x] Based your PR on the latest version of the correct branch (master or 3.x)
* [x] Followed [PEP8](https://www.python.org/dev/peps/pep-0008/) for code style
* [x] Ensured your pull request hasn't eliminated unrelated blank lines/spaces,
      modified the ``spyder/defaults`` directory, or added new icons/assets
* [x] Wrote at least one-line docstrings for any new functions
* [x] Added at least one unit test covering the changes, if at all possible
* [x] Described your changes and the motivation for them below
* [x] Noted what issue(s) this pull request resolves, creating one if needed
* [x] Included a screenshot, if this PR makes any visible changes to the UI

## Description of Changes

Add support for f-strings

### Issue(s) Resolved

Fixes #7265

### CODE REVIEW

**Comment by dalthviz** [comment]
@ccordoba12 I thought the idea was to have a new option that when activated detects f-strings. Should this functionality always be active (if the Python version is 3.6+)?

File: spyder/widgets/sourcecode/codeeditor.py [hunk 1]
```diff
@@ -2624,6 +2632,18 @@ def __in_comment(self):
         else:
             return False
 
+    def __in_string(self):
+        """Check if we are in a string."""
+        if self.highlighter:
+            current_color = self.__get_current_color()
+            comment_color = self.highlighter.get_color_name('string')
```
Comment (jitseniesen) [inline spyder/widgets/sourcecode/codeeditor.py]: why is this called comment_color ?

Comment (dalthviz) [reply spyder/widgets/sourcecode/codeeditor.py]: I took the lines from the `__in_comment` method and forgot to change the name of the variable. Thank you for noticing it :+1:

**Comment by dalthviz** [comment]
A preview:
f

**Comment by jitseniesen** [comment]
I can imagine that this feature is annoying for some users. With this enabled, if you write code like `if char == '{'` or `'{}'.format(var)`, an f is automatically added. That's a good argument for adding an option.

If an option is added, it should probably have a better description like "Automatically insertion of 'f' in front of f-strings" (as I did not at all understand what it was about before I read the issue report) and be grouped in the "Advanced" section with the other automatic insertion options.

I can't find the test(s) claimed to be present by the checklist at the top.

**Comment by ccordoba12** [comment]
We decided with @dalthviz to not be so clever: adding an `f` won't do anything additional, and users will have to prefix their strings with an `f` to get completions insider them.

**Comment by dalthviz** [comment]
A new preview:
f

**Comment by ccordoba12** [comment]
Let's close this because it depends on the old completion architecture, so it can't easily br ported to the new one.

**Pull request closed without merging**